modal/mount.py

```diff
@@ -547,7 +562,9 @@ async def _put_file(file_spec: FileUploadSpec) -> api_pb2.MountFile:
                 logger.debug(
                     f"Uploading file {file_spec.source_description} to {remote_filename} ({file_spec.size} bytes)"
                 )
-                request2 = api_pb2.MountPutFileRequest(data=file_spec.content, sha256_hex=file_spec.sha256_hex)
+                # Read content lazily only when needed for upload
+                content = await asyncio.get_event_loop().run_in_executor(None, file_spec.get_content)
```

Comment:
Nit: I think this ends up doing the same thing:

```suggestion
                content = await asyncio.to_thread(file_spec.get_content)
```

Reply:
Do you mean there's overhead to doing the "check if we already have content" part on the other thread?